src/gc/gcinterface.h

```diff
@@ -451,6 +464,23 @@ class IGCHeap {
     // Walks an object, invoking a callback on each member.
     virtual void WalkObject(Object* obj, walk_fn fn, void* context) = 0;
 
+    // Walk the heap object by object.
+    virtual void WalkHeap(walk_fn fn, void* context, int gen_number, BOOL walk_large_object_heap_p) = 0;
+    
+    // Walks the survivors and get the relocation information if objects have moved.
+    virtual void WalkSurvivors(void* gc_context, record_surv_fn fn, size_t diag_context, walk_surv_type type) = 0;
```

Comment:
`diag_context` should be `void*` - to match what is used for context everywhere else.